doc/contributors/design-proposals/pull-bundle-on-a-cluster.md

```diff
@@ -0,0 +1,127 @@
+# Pull Bundle on a Cluster
+
+The purpose of this proposal is to allow bundle data to be pulled at install time. At a high level, this starts with launching the bundle inside a pod and then writing the data found inside in the container into a configmap.
+
+## Launching a bundle image for data extraction
+
+An API will be made that handles launching a job. The only required parameter will be specifying the bundle image.
+
+The job spec will look similar to the following:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: deploy-bundle-image
+spec:
+  #ttlSecondsAfterFinished: 0 # alpha feature - https://github.com/kubernetes/enhancements/issues/592, https://github.com/kubernetes/kubernetes/pull/82082
+  template:
+    metadata:
+      name: bundle-image
+    spec:
+      containers:
+      - name: bundle-image
+        image: &image bundle-image
+        command: ['/injected/operator-registry', 'serve']
+        env:
+          - name: CONTAINER_IMAGE
+            value: *image
+        volumeMounts:
+        - name: copydir
+          mountPath: /injected
+      initContainers:
+      - name: copy-binary
+        image: init-operator-manifest
+        imagePullPolicy: Never
+        command: ['/bin/cp', '/operator-registry', '/copy-dest']
+        volumeMounts:
+        - name: copydir
+          mountPath: /copy-dest
+      volumes:
+      - name: copydir
+        emptyDir: {}
+      restartPolicy: OnFailure
+```
+
+## Serving the bundle data
+
+New code will be made in operator-registry to provide functionality for traversing the directies of the bundle image for writing to a configmap (example above is "operator-registry serve"), the format for which is discussed in more detail below. The configmap will also use a generated name to avoid collisions and will be labeled to match the requested bundle image. It will be the responsibility of the caller to 1) launch the job, 2) watch for the target configmap to be created 3) to delete the job (until ttlSecondsAfterFinished is available) 4) delete the configmap after reading.
```

Comment:
> 3) to delete the job (until ttlSecondsAfterFinished is available)

Is there any reason the caller can't just delete the `job` when it reaches `Completed`? Why wait?

> 4) delete the configmap after reading.

I think we want to keep the `ConfigMap` around as an on-cluster cache.

Reply:
I think (3) is:

- the caller will be responsible for deleting the job
- when ttlSecondsAfterfinished is available, the caller will no longer be required to delete the job.

Reply:
Yeah, I was referring to the ttlSecondAfterFinished feature being non-alpha. I've tried to rewrite the proposal in such a way that makes it clear that deletion is up to the caller.